### DISCUSSION TITLE
Repeat Visits to Node

### DISCUSSION THREAD

**Discussion by authman** [google/or-tools#2921] [Routing (and legacy CP) questions] [answered]
Running classic Vehicle Routing Problem of 3 vehicles, 30 nodes. However, out of the 30 node, I would like to enforce that 6 of those nodes are visited by **all three** vehicles. Is there a way to relax the constraint that each node is only visited by a single vehicle _just_ for this subset of nodes, leaving it in place for the remaining 24 nodes?

Much appreciated

**Comment by Mizux** [comment, marked as answer]
No, you need to duplicate these nodes  + you can force that each duplicate (and the original) are visited by a specific vehicle index thus ensuring each duplicate is visited by a different vehicle.

e.g. node 31, duplicate as node 42 and 43 then we can do:
```python
index = manager.NodeToIndex(31)
routing.VehicleVar(index).SetValue(0)

index = manager.NodeToIndex(42)
routing.VehicleVar(index).SetValue(1)

index = manager.NodeToIndex(43)
routing.VehicleVar(index).SetValue(2)
```
So node 31 can onlybe visited by vehicle 0, node 42 by vehicle 1 and node 43 by vehicle 2...
